contributors/design-proposals/scheduling/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,187 @@
+
+Thus, we decide to reuse the existing `Scopes` of `ResourceQuotaSpec` to provide a richer semantics for quota to cooperate with priority classes.
+
+## Overview
+
+This design doc introduces how to define a group of priority class scopes for the quota to match with and explains how quota enforcement logic is changed to apply the quota to pods with the given priority classes.
+
+## Detailed Design
+
+### Changes in ResourceQuota
+
+ResourceQuotaSpec contains an array of filters, `Scopes`, that if mentioned, must match each object tracked by a ResourceQuota.
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+        ...
+
+        // A collection of filters that must match each object tracked by a quota.
+        // If not specified, the quota matches all objects.
+        // +optional
+        Scopes []ResourceQuotaScope
+}
+```
+Four new `ResourceQuotaScope` will be defined for matching pods based on priority class names.
+```go
+// A ResourceQuotaScope defines a filter that must match each object tracked by a quota
+type ResourceQuotaScope string
+
+const (
+        ...
+        ResourceQuotaScopePriorityClassNameExists ResourceQuotaScope = "PriorityClassNameExists"
+        // Match all pod objects that do not have any priority class mentioned
+        ResourceQuotaScopePriorityClassNameNotExists ResourceQuotaScope = "PriorityClassNameNotExists"
+        // Match all pod objects that have priority class from the set
+        ResourceQuotaScopePriorityClassNameIn ResourceQuotaScope = "PriorityClassNameIn"
+        // Match all pod objects that do not have priority class from the set
+        ResourceQuotaScopePriorityClassNameNotIn ResourceQuotaScope = "PriorityClassNameNotIn"
+)
+```
+
+### Changes in Admission Controller Configuration
+
+A new field `MatchScopes` will be added to `Configuration.LimitedResource`. `MatchScopes` will be a collection of one or more of the four newly added priority class based `Scopes` that are explained in above section.
+
+```go
+// Configuration provides configuration for the ResourceQuota admission controller.
+type Configuration struct {
+        ...
+        LimitedResources []LimitedResource
+}
+
+// LimitedResource matches a resource whose consumption is limited by default.
+// To consume the resource, there must exist an associated quota that limits
+// its consumption.
+type LimitedResource struct {
+        ...
+
+        // MatchScopes is a collection of filters based on priority classes.
+        // If the object in the intercepted request matches these rules,
+        // quota system will ensure that corresponding quota MUST have
+        // priority based Scopes matches the object in request. 
+        //
+        // If MatchScopes has matched on an object, request will be denied if there is no quota
```

Comment:
s/request will be denied if there is no quota matches priority class based Scopes./request for the resource will be denied if there is no quota with matching Scopes."